docs/OpenShift GitOps Usage Guide.md

```diff
@@ -0,0 +1,1291 @@
+
+### Login via Keycloak (RHSSO)
+
+![image alt text](assets/image_21.png)
+
+### **Keycloak Identity brokering with OpenShift OAuthClient**
+
+Prior to configuring OpenShift 4 Identity Provider, please locate the correct OpenShift 4 API URL up. The easiest way to obtain it is to invoke the following command (this might require installing jq command separately) 
+
+`curl -s -k -H "Authorization: Bearer $(oc whoami -t)" https://<openshift-user-facing-api-url>/apis/config.openshift.io/v1/infrastructures/cluster | jq ".status.apiServerURL". `
+
+In most cases, the address will be protected by HTTPS. Therefore, it is essential to configure X509_CA_BUNDLE in the container and set it to /var/run/secrets/kubernetes.io/serviceaccount/ca.crt. Otherwise, Keycloak won’t be able to communicate with the API Server.
+
+Go to Identity Providers and select Openshift v4. Set the Base Url to OpenShift 4 API URL. Client ID as keycloak-broker(can be anything) and Client Secret to any secret that you want to set in Step 7.
+
+## **Registering an OAuth Client** 
+
+```
+kind: OAuthClient
+apiVersion: oauth.openshift.io/v1
+metadata:
+ name: keycloak-broker 
+secret: "..." 
+redirectURIs:
+"https://keycloak-keycloak.apps.dev-svc-4.7-020201.devcluster.openshift.com/auth/realms/myrealm/broker/openshift-v4/endpoint" 
+grantMethod: prompt 
+```
+
+1. The name of the OAuth client is used as the client_id parameter when making requests to <namespace_route>/oauth/authorize and <namespace_route>/oauth/token.\
+2. The secret is used as the client_secret parameter when making requests to <namespace_route>/oauth/token.\
+3. The redirect_uri parameter specified in requests to <namespace_route>/oauth/authorize and <namespace_route>/oauth/token must be equal to or prefixed by one of the URIs listed in the redirectURIs parameter value. Refer to the screenshot in Step 6. \
+4. The grantMethod is used to determine what action to take when this client requests tokens and has not yet been granted access by the user. Specify auto to automatically approve the grant and retry the request, or prompt to prompt the user to approve or deny the grant.
+
+
+
+Now you can log into Argo CD using your Openshift Credentials through Keycloak as an Identity Broker.
+
+
+
+![image alt text](assets/image_22.png)
+
+**Configure Groups and ArgoCD RBAC**
+
+After this point, You must provide relevant access to the user to create applications, projects e.t.c.,. This can be achieved in two steps.
+
+1. Add the logged in user to the keycloak group *ArgoCDAdmins* created earlier. 
+
+![image alt text](assets/image_23.png)
+
+2. Make sure *ArgoCDAdmins *group has required permissions in the argocd-rbac configmap. Please refer to the ArgoCD Rbac documentation [here](https://argoproj.github.io/argo-cd/operator-manual/rbac/).
+
+In this case, I wish to provide admin access to the users in *ArgoCDAdmins *groupargocd-rbac-cmArgoCDAdmins
```

Comment:
remove space after ArgoCDAdmins, also not sure what the ending of this sentence should be but it definitely got messed up